libsplinter/src/rest_api/auth/rbac/rest_api/actix_web_1/assignments.rs

```diff
@@ -0,0 +1,533 @@
+
+        role_based_auth_store
+            .add_role(role)
+            .expect("Unable to add role");
+
+        let assignment = AssignmentBuilder::new()
+            .with_identity(Identity::Key("x".into()))
+            .with_roles(vec!["role-1".to_string(), "role-2".to_string()])
+            .build()
+            .expect("Unable to build assignment");
+
+        role_based_auth_store
+            .add_assignment(assignment)
+            .expect("Unable to add assignment");
+
+        let assignment = AssignmentBuilder::new()
+            .with_identity(Identity::User("y".into()))
+            .with_roles(vec!["role-1".to_string(), "role-2".to_string()])
+            .build()
+            .expect("Unable to build assignment");
+
+        role_based_auth_store
+            .add_assignment(assignment)
+            .expect("Unable to add assignment");
+
+        let (shutdown_handle, join_handle, bind_url) =
+            run_rest_api_on_open_port(vec![make_assignments_resource(Box::new(
+                role_based_auth_store,
+            ))]);
+
+        let url = Url::parse(&format!("http://{}/authorization/assignments", bind_url))
+            .expect("Failed to parse URL");
+
+        let resp = Client::new()
+            .get(url)
+            .header(
+                "SplinterProtocolVersion",
+                protocol::AUTHORIZATION_PROTOCOL_VERSION,
+            )
+            .send()
+            .expect("Failed to perform request");
+
+        assert_eq!(resp.status(), StatusCode::OK);
+        let body: JsonValue = resp.json().expect("Failed to deserialize body");
+
+        let json_assignments = get_in!(body, &["data"], as_array)
+            .expect("data field is not an array")
+            .to_vec();
+
+        assert_eq!(2, json_assignments.len());
+
+        assert_eq!(
+            &json!({
+                "identity": "x",
+                "identity_type": "key",
+                "roles": ["role-1", "role-2"],
+            }),
+            json_assignments.get(0).expect("no first item")
+        );
+
+        assert_eq!(
+            &json!({
+                "identity": "y",
+                "identity_type": "user",
+                "roles": ["role-1", "role-2"],
+            }),
+            json_assignments.get(1).expect("no second item")
+        );
+
+        assert_eq!(
+            body.get("paging").expect("no paging field in response"),
+            &to_value(create_test_paging_response(
+                0,
+                100,
+                0,
+                0,
+                0,
+                2,
+                "/authorization/assignments?"
+            ))
+            .expect("failed to convert expected paging")
+        );
+
+        shutdown_handle
+            .shutdown()
+            .expect("Unable to shutdown rest api");
+        join_handle.join().expect("Unable to join rest api thread");
+    }
+
+    /// Tests a GET /authorization/assignments request which returns the set of assignments.
```

Comment:
The comments here aren't sufficient to be able to judge whether the test against the intent of the test.

Reply:
Updated all of the test comments